File: 102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if root:
            q = [root]
        else:
            return []
        level = []
        while q:
            l, bfs = [], []
            for node in q:
                ###l will get the all values in the level.
                l.append(node.val)
                if node.left:
                    bfs.append(node.left)
                if node.right:
                    bfs.append(node.right)
            ###Always get the level in every itervtion.
            q = bfs
            level.append(l)
        return level

File: 102. Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (root == nullptr){
            return {};
        }
        vector<vector<int>>res;
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()){
            vector<int> tmp;
            int q_size = q.size();
            while (q_size){
                TreeNode* currNode = q.front();
                q.pop();
                tmp.push_back(currNode->val);
                if (currNode->left){
                    q.push(currNode->left);
                }
                if (currNode->right){
                    q.push(currNode->right);
                }
                q_size--;
            }
            res.push_back(tmp);
            
        }
        return res;
        
    }

};
